clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -968,6 +972,23 @@ static bool pathOnlyHandlesGslPointer(IndirectLocalPath &Path) {
   return false;
 }
 
+static bool isAssginmentOperatorLifetimeBound(CXXMethodDecl *CMD) {
+  if (!CMD)
+    return false;
+  assert(CMD->getOverloadedOperator() == OverloadedOperatorKind::OO_Equal);
```

Comment:
What about compound assignment operators? Do you plan to support those in a follow-up?

That being said, I'd love to see a test case just to make sure this assert is not triggered.